### PR TITLE
Fix epJSON Schema cmake dependency error for embedded one

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------
 - Fixes #ISSUENUMBERHERE (IF THIS IS A DEFECT)
 - DESCRIBE PURPOSE OF THIS PULL REQUEST

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: scripts/dev/generate_embeddable_epJSON_schema/generate_embedded_epJSON_schema.cmake [hunk 1]
```diff
@@ -1,5 +1,11 @@
-execute_process( COMMAND "${EnergyPlus_embeddable_epJSON_schema}" "${EnergyPlus_RUNTIME_OUTPUT_DIRECTORY}/Energy+.schema.epJSON" TIMEOUT 90 RESULT_VARIABLE generate_embedded_epJSON_schema OUTPUT_VARIABLE embedded_epJSON_schema)
-if( ${generate_embedded_epJSON_schema} MATCHES ".*timeout.*" )
+execute_process(
+  COMMAND "${EnergyPlus_embeddable_epJSON_schema}" "${EnergyPlus_RUNTIME_OUTPUT_DIRECTORY}/Energy+.schema.epJSON"
+  TIMEOUT 90
+  RESULT_VARIABLE generate_embedded_epJSON_schema
+  OUTPUT_VARIABLE embedded_epJSON_schema
+)
+message(AUTHOR_WARNING "generating embedded epJSON schema")
```
Comment (jmarrec) [inline scripts/dev/generate_embeddable_epJSON_schema/generate_embedded_epJSON_schema.cmake]: Just adding an author_warning here

File: scripts/dev/generate_epJSON_schema/generate_epJSON_schema.py [hunk 2]
```diff
@@ -59,7 +59,7 @@
 import sys
 from os import path
 
-
+print("Generating the epJSON schema!")
```
Comment (jmarrec) [inline scripts/dev/generate_epJSON_schema/generate_epJSON_schema.py]: And a print statement here

File: idd/Energy+.idd.in [hunk 3]
```diff
@@ -397,6 +397,9 @@ Version,
   A1 ; \field Version Identifier
       \default ${CMAKE_VERSION_MAJOR}.${CMAKE_VERSION_MINOR}
 
+FakeObject,
+  A1; \field Name
+
```
Comment (jmarrec) [inline idd/Energy+.idd.in]: Fake change. If everything worked well, I would expect to the see both the epJSON schema and the author warning for the embedded one, but the latter is missing

```diff
[  5%][1/19] Generating ../../Products/Energy+.schema.epJSON
Generating the epJSON schema!
- [ 10%][2/19] Generating EmbeddedEpJSONSchema.cc
-CMake Warning (dev) at /Users/julien/Software/Others/EnergyPlus/scripts/dev/generate_embeddable_epJSON_schema/generate_embedded_epJSON_schema.cmake:7 (message):
-  generating embedded epJSON schema
-This warning is for project developers.  Use -Wno-dev to suppress it.
```

**Comment by jmarrec** [comment]
Before: https://github.com/NREL/EnergyPlus/actions/runs/5334849873/jobs/9667205583#step:6:64

> AssertionError: Failed after IDD change: missing: {'generating embedded epJSON schema'}

After: all good

https://github.com/NREL/EnergyPlus/actions/runs/5334869904/jobs/9667258575

File: .github/workflows/epjson.yml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+            f.write('\n'.join(lines) + '\n')
+
+        lines = build()
+        ensure_target_built(lines, "Failed after IDD change")
+
+        with open(IDD_IN, 'w') as f:
+            f.write('\n'.join(ori_lines) + '\n')
+
+        lines = build()
+        ensure_target_built(lines, "Failed after IDD change revert")
```
Comment (jmarrec) [inline .github/workflows/epjson.yml]: A test workflow

File: src/EnergyPlus/CMakeLists.txt [hunk 5]
```diff
@@ -729,7 +729,11 @@ add_custom_command(
         -D "EnergyPlus_embeddable_epJSON_schema:PATH=$<TARGET_FILE:generate_embeddable_epJSON_schema>"
         -P "${PROJECT_SOURCE_DIR}/scripts/dev/generate_embeddable_epJSON_schema/generate_embedded_epJSON_schema.cmake"
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-  DEPENDS generate_embeddable_epJSON_schema epjson_schema "${CMAKE_CURRENT_SOURCE_DIR}/InputProcessing/EmbeddedEpJSONSchema.in.cc"
+  DEPENDS
+    generate_embeddable_epJSON_schema
+    "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/Energy+.schema.epJSON" # File-level dependency
+    epjson_schema                                             # target-level dependency, to avoid race conditions
+    "${CMAKE_CURRENT_SOURCE_DIR}/InputProcessing/EmbeddedEpJSONSchema.in.cc"
```
Comment (jmarrec) [inline src/EnergyPlus/CMakeLists.txt:736]: the actual fix

File: CMakeLists.txt [hunk 6]
```diff
@@ -1,3 +1,5 @@
+cmake_minimum_required(VERSION 3.17)
```
Comment (Myoldmopar) [inline CMakeLists.txt:1]: OK cool, bumping this earlier.

File: src/EnergyPlus/CMakeLists.txt (hunk 5 above)
Comment (Myoldmopar) [reply src/EnergyPlus/CMakeLists.txt:736]: Thanks! I'll test it out.

**Comment by jmarrec** [comment]
Superseded by #10091

**Pull request closed without merging**